File: euler_problem3.py
import math

def isPrime(number):
	i =2
	while i <math.ceil(math.sqrt(number)):
		if number%i == 0:
			return False
		i += 1
	return True


def euler_problem3(number):
	j = number-1
	while j >1:
		if number%j == 0:
			if isPrime(j):
				return j
print(euler_problem3(600851475143))

File: euler_problem3.java
import java.lang.*;

public class euler_problem3 {


	public boolean isPrime(double number) {
		for (double i =2; i< Math.ceil(Math.sqrt(number)); i++) {
			if (number % i == 0 ) return false;
		}
		return true;
	}
	public static void main (String[] args) {
		
		euler_problem3 euler_problem3 = new euler_problem3();
		for (double n = 600851475142L; n>1; n--) {
			if (600851475143L % n==0 )
				System.out.print(n + "        "); 
				if (euler_problem3.isPrime(n)) {
					System.out.println(n);
					return;
				}
		}
		
	}

}
